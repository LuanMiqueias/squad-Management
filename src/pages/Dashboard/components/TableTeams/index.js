import React from "react";
import styles from "./style.module.css";

import iconList from "../../../../assets/icons/arrowList.svg";
import iconDelete from "../../../../assets/icons/delete.svg";
import iconShare from "../../../../assets/icons/share.svg";
import iconEdit from "../../../../assets/icons/edit.svg";
import { Link } from "react-router-dom";

const TableTeams = () => {
  const [dataSorted, setDataSorted] = React.useState([]);
  const [isDescSort, setIsDescSort] = React.useState(true);

  React.useEffect(() => {
    if (localStorage.teams) {
      const dataLocal = JSON.parse(localStorage.teams);
      setDataSorted([...dataLocal.teams]);
    }
    //eslint-disable-next-line
  }, []);
  const orderBy = React.useCallback(
    (type) => {
      const arraySorted = dataSorted.sort((a, b) => {
        if (!isDescSort) {
          setIsDescSort(!isDescSort);
          if (a[type].toLowerCase() < b[type].toLowerCase()) {
            return -1;
          }
        } else {
          setIsDescSort(!isDescSort);
          if (a[type].toLowerCase() > b[type].toLowerCase()) {
            return -1;
          }
        }
        return 0;
      });
      if (dataSorted) {
        setDataSorted([...arraySorted]);
      }
    },
    [isDescSort, dataSorted]
  );
  function deleteTeam(id) {
    const dataLocal = JSON.parse(localStorage.teams);
    const filterDataLocal = dataLocal.teams.filter((data) => data.id !== id);
    setDataSorted(filterDataLocal);
    localStorage.teams = JSON.stringify({
      teams: [...filterDataLocal],
    });
  }
  return (
    <div className={styles.table}>
      <div className={styles.content}>
        <span
          className={`${styles.TableHeader} ${styles.headerName}`}
          onClick={() => {
            orderBy("name");
          }}
          aria-label="name"
        >
          Name
          <img src={iconList} alt="" className="icon" />
        </span>
        <span
          className={`${styles.TableHeader} ${styles.headerDescripition}`}
          onClick={() => {
            orderBy("description", dataSorted);
          }}
          aria-label="description"
        >
          Description
          <img src={iconList} alt="" className="icon" />
        </span>
      </div>
      {dataSorted.map(({ name, description, id }) => {
        return (
          <div className={styles.content} key={`${name}`} aria-label={name}>
            <span className={styles.TableItems}>{name}</span>
            <span className={`${styles.TableItems} ${styles.containerIcons}`}>
              <span>{description}</span>
              <div className={styles.contentIcons}>
                <button
                  className={styles.iconBtn}
                  onClick={() => deleteTeam(id)}
                  aria-label="Delete"
                >
                  <img src={iconDelete} alt="" className="icon" />
                  <div className={styles.tableTooltip}>Delete</div>
                </button>
                <button className={styles.iconBtn} aria-label="Share">
                  <img src={iconShare} alt="" className="icon" />
                  <div className={styles.tableTooltip}>Share</div>
                </button>
                <Link
                  to={`/edit/${id}`}
                  className={styles.iconBtn}
                  aria-label="Edit"
                >
                  <img src={iconEdit} alt="" className="icon" />
                  <div className={styles.tableTooltip}>Edit</div>
                </Link>
              </div>
            </span>
          </div>
        );
      })}
    </div>
  );
};

export default TableTeams;
