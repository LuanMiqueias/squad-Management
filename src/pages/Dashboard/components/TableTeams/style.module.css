.table{
  padding: 24px 8px;
  font-size: .88rem;
  font-weight: 600;
  color:var(--black);
  flex:1;
}
.TableHeader{
  padding: 24px 0 ;
  display: flex;
  align-items: flex-start;
  justify-content: space-between;
  cursor: pointer;
  transition: all .05s;
  user-select: none;
}
.TableHeader:hover{
  opacity: .8;
}
.headerName{
  border-right: 1px solid var(--divider);
}
.content{
  display: flex;
  gap:24px;
  align-items: center;
  padding: 12px 8px 12px 24px;
  transition: all .05s;

}
.content:nth-child(1){
  border:0;
  padding: 8px 8px 8px 24px;  
  margin-bottom: 16px;
}
.content + .content + .content{
  border-top: 1px solid var(--divider);
}
.content + .content:hover{
  color:var(--hover-text-color);
  background: var(--hover-background);
  border-color: transparent;
  cursor: pointer;
  border-radius: 5px;
}
.content:hover + .content{
  border-color: transparent;
}
.content span:nth-child(1){
  flex: .6;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}
.content span{
  flex: 2;
  text-align: left;
  padding: 0 16px 0 0 ;  
  text-transform: capitalize;

}
.containerIcons{
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
}

.contentIcons{
  display: flex;
  gap:4px;
  height: 100%;
}

.iconBtn{
  display: flex;
  align-items: center;
  justify-content: center;
  border:0;
  background: none;
  outline: none;
  transition: all .1s;
  padding: 8px 6px;
  height: 40px;
  opacity: 0;
  transform: translateY(25%);
  transition: transform .2s;
  position: relative;
  z-index: 1000;
}
.content + .content:hover .iconBtn{
  opacity: 1;
  transform: translateY(0%);
}
.iconBtn:hover.iconBtn img{
  transform: translateY(-25%) scale(2);
}
.iconBtn img{
  height: .88rem;
  transition: all .1s;
}

.tableTooltip{
  display: none;
  position: absolute;
  white-space: nowrap;
  background: var(--black);
  border-radius: 5px;
  padding: 8px 16px;
  font-size: .75rem;
  color:var(--white);
  animation: animaAriaLabel .1s forwards;
  top:-100%;
  align-items: center;
  justify-content: center;
  transform: translateX(-25%);
  text-transform: initial;
}
.tableTooltip::before {
  content:'';
  position: absolute;
  width: 10px;
  height: 10px;
  background: var(--black);
  border-radius: 3px;
  color:var(--white);
  transform: translateY(150%) translateX(175%) rotate(45deg);
}
.iconBtn:hover .tableTooltip{
  display: flex;
}