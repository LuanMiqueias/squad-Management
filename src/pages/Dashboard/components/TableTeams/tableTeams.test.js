import { render } from "@testing-library/react";
import TableTeams from ".";

const props = [
  {
    name: "test 1",
    description: "description 1",
  },
  {
    name: "teste 2",
    description: "Description 1",
  },
];

it("should return name and description", () => {
  const { getAllByLabelText } = render(<TableTeams data={props} />);
  const names = getAllByLabelText("name");
  const descriptions = getAllByLabelText("description");

  names.forEach((name) => {
    return expect(name).not.toHaveTextContent("");
  });
  descriptions.forEach((description) => {
    return expect(description).not.toHaveTextContent("");
  });
});

it("should match snapshot", () => {
  const { container } = render(<TableTeams data={props} />);
  expect(container.firstChild).toMatchSnapshot();
});
