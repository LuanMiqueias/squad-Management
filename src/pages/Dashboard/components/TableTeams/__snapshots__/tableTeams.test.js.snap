// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  class="table"
>
  <div
    class="content"
  >
    <span
      class="TableHeader headerName"
    >
      Name
      <img
        alt=""
        class="icon"
        src="arrowList.svg"
***REMOVED***
    </span>
    <span
      class="TableHeader headerDescripition"
    >
      Description
      <img
        alt=""
        class="icon"
        src="arrowList.svg"
***REMOVED***
    </span>
  </div>
  <div
    class="content"
  >
    <span
      aria-label="name"
      class="TableItems"
    >
      test 1
    </span>
    <span
      aria-label="description"
      class="TableItems containerIcons"
    >
      description 1
      <div
        class="contentIcons"
      >
        <button
          class="iconBtn"
        >
          <img
            alt=""
            class="icon"
            src="delete.svg"
    ***REMOVED***
        </button>
        <button
          class="iconBtn"
        >
          <img
            alt=""
            class="icon"
            src="share.svg"
    ***REMOVED***
        </button>
        <button
          class="iconBtn"
        >
           
          <img
            alt=""
            class="icon"
            src="edit.svg"
    ***REMOVED***
        </button>
      </div>
    </span>
  </div>
  <div
    class="content"
  >
    <span
      aria-label="name"
      class="TableItems"
    >
      teste 2
    </span>
    <span
      aria-label="description"
      class="TableItems containerIcons"
    >
      Description 1
      <div
        class="contentIcons"
      >
        <button
          class="iconBtn"
        >
          <img
            alt=""
            class="icon"
            src="delete.svg"
    ***REMOVED***
        </button>
        <button
          class="iconBtn"
        >
          <img
            alt=""
            class="icon"
            src="share.svg"
    ***REMOVED***
        </button>
        <button
          class="iconBtn"
        >
           
          <img
            alt=""
            class="icon"
            src="edit.svg"
    ***REMOVED***
        </button>
      </div>
    </span>
  </div>
</div>
`;
